pandas/core/indexes/accessors.py

```diff
@@ -324,7 +324,9 @@ def __new__(cls, data):
 
         orig = data if is_categorical_dtype(data) else None
         if orig is not None:
-            data = Series(orig.values.categories, name=orig.name, copy=False)
+            data = Series(
+                orig.values.astype("datetime64[ns]"), name=orig.name, copy=False
```

Comment:
why the astype?  what if we get here with e.g. timedelta64 or Period?

Reply:
Agreed, it was just a quick attempt based on what I'd read in the issue without me having fully understood the issue, sorry for having put this up so soon

Reply:
No sweat.  ping me when this is ready for review

Reply:
@jbrockmendel thanks - I'm happy to have this reviewed, whenever you have time